js/bitmart.js

```diff
@@ -500,141 +500,100 @@ module.exports = class bitmart extends Exchange {
-            const contract = this.safeValue (market, 'contract', {});
-            const id = this.safeString (contract, 'contract_id');
-            const numericId = this.safeInteger (contract, 'contract_id');
-            const baseId = this.safeString (contract, 'base_coin');
-            const quoteId = this.safeString (contract, 'quote_coin');
-            const settleId = this.safeString (contract, 'price_coin');
+        for (let i = 0; i < tickers.length; i++) {
+            const market = tickers[i];
+            const id = this.safeString (market, 'contract_symbol');
+            const baseId = id.slice (0, -4);
+            const quoteId = id.slice (-4);
             const base = this.safeCurrencyCode (baseId);
             const quote = this.safeCurrencyCode (quoteId);
-            const settle = this.safeCurrencyCode (settleId);
-            //
-            // https://github.com/bitmartexchange/bitmart-official-api-docs/blob/master/rest/public/symbols_details.md#response-details
-            // from the above API doc:
-            // quote_increment Minimum order price as well as the price increment
-            // price_min_precision Minimum price precision (digit) used to query price and kline
-            // price_max_precision Maximum price precision (digit) used to query price and kline
-            //
-            // the docs are wrong: https://github.com/ccxt/ccxt/issues/5612
-            //
-            const amountPrecision = this.safeNumber (contract, 'vol_unit');
-            const pricePrecision = this.safeNumber (contract, 'price_unit');
-            const contractType = this.safeValue (contract, 'contract_type');
+            const splitId = id.split (id, '_');
+            const splitIdEnd = this.safeString (splitId, 1);
+            let settle = 'USDT';
+            let symbol = base + '/' + quote + ':' + settle;
+            let type = 'swap';
+            let swap = true;
             let future = false;
-            let swap = false;
-            let type = 'contract';
-            let symbol = base + '/' + quote;
-            const expiry = this.parse8601 (this.safeString (contract, 'delive_at'));
-            if (contractType === 1) {
-                type = 'swap';
-                swap = true;
-                symbol = symbol + ':' + settle;
-            } else if (contractType === 2) {
-                type = 'future';
-                future = true;
-                symbol = symbol + ':' + settle + '-' + this.yymmdd (expiry, '');
+            if (splitIdEnd !== undefined) {
+                settle = 'BTC';
+                symbol = base + '/' + quote + ':' + settle;
+                if (splitIdEnd !== 'PERP') {
+                    const date = this.iso8601 (this.milliseconds ());
+                    const splitDate = this.split (date, '-');
+                    const year = splitDate[0];
+                    const shortYear = year.slice (0, 2);
+                    const expiry = shortYear + splitIdEnd;
```

Comment:
This expiry is in the form `yymmdd` but I don't think we have a way to convert this to a timestamp or `iso8601`

Reply:
Let's do this:

```JavaScript
const expiryMonth = splitEnd.slice (0, 2);
const expiryDay = splitEnd.slice (2, 4);
expiry = this.parse8601 (year + '-' + expiryMonth + '-' + expiryDay + 'T00:00:00Z');
```